code/controllers/subsystem/processing/resinshaping.dm

```diff
@@ -48,6 +48,9 @@ SUBSYSTEM_DEF(resinshaping)
 	var/player_key = "[the_builder.client?.ckey]"
 	if(!player_key)
 		return 0
+	var/turf/selectedturf = get_turf(the_builder)
+	if(selectedturf.get_build_prereqs())
+		return
```

Comment:
Why is this being checked here?

Reply:
This is a check to avoid cheating the quickbuild system, if it did not exist you would be able to build normally (which doesn't cost quickbuild points) and then deconstruct to gain infinite points.

Reply:
this is just the decrement proc though? Its a bit janky to have it here, considering this is called when building, not just deconstructing.

Reply:
Um, I'm willing to be wrong here, but according to VSC, the only things calling decrement_build_counter is attack_alien on harm intent.

Reply:
oh you're right, my bad. This could still get niche issues like building before silo outside caves, silo dropping nearby, and removing the walls refunding points however.